src/requested_path.rs

```diff
@@ -43,12 +44,7 @@ impl RequestedPath {
         let index_path = self.path.join("index.html");
 
         match fs::metadata(&index_path) {
-            Ok(m) =>
-                if m.is_file() {
-                    Some(index_path)
-                } else {
-                    None
-                },
+            Ok(m) if m.is_file() => Some(index_path),
```

Comment:
I believe this will work if you use `Ok(ref m)`. You can't move out of the target of a match in a pattern in a conditional.